sleap_roots/pipelining.py

```diff
@@ -0,0 +1,625 @@
+)
+from sleap_roots.convhull import get_convhull_features
+from sleap_roots.ellipse import fit_ellipse
+from sleap_roots.networklength import get_network_distribution
+from sleap_roots.networklength import get_network_distribution_ratio
+from sleap_roots.networklength import get_network_solidity
+from sleap_roots.networklength import get_network_width_depth_ratio
+from sleap_roots.scanline import count_scanline_intersections
+from sleap_roots.tips import get_tips, get_tips_percentile
+from typing import Tuple
+
+
+def get_statistics(
+    traits_array: np.ndarray,
+) -> Tuple[float, float, float, float, float]:
+    """Get 5 statistics of the traits.
+
+    Args:
+        traits_array: the traits array that need to calculate 5 statistics.
+
+    Returns:
+        A Tuple of 5 statistics: max, min, mean, std, median.
+    """
+    trait_max = np.nanmax(traits_array)
+    trait_min = np.nanmin(traits_array)
+    trait_mean = np.nanmean(traits_array)
+    trait_std = np.nanstd(traits_array)
+    trait_median = np.nanmedian(traits_array)
+    return trait_max, trait_min, trait_mean, trait_std, trait_median
+
+
+def get_pts_pr_lr(h5, rice=False, frame=0, primaryroot=True):
+    """Get all predicted points of primary/lateral root (# instance, # nodes, 2).
+
+    Args:
+        h5: h5 file, SLEAP prediction.
+        rice: Boolean value, where true is rice (default), false is other species.
+        frame: the frame index, default value is 0, i.e., first frame.
+        primaryroot: Boolean value, where true is primary root, false is lateral root.
+
+    Returns:
+        An array of points location in shape of (# points, 2).
+    """
+    if rice:
+        series = Series.load(
+            h5, primary_name="longest_3do_6nodes", lateral_name="main_3do_6nodes"
+        )
+    else:
+        series = Series.load(
+            h5, primary_name="primary_multi_day", lateral_name="lateral_3_nodes"
+        )
+    primary, lateral = series[frame]
+    if primaryroot:
+        return primary.numpy()
+    else:
+        return lateral.numpy()
+
+
+def get_pts_all(h5, rice=False, frame=0) -> np.ndarray:
+    """Get all predicted points in shape of (# points, 2).
+
+    Args:
+        h5: h5 file, SLEAP prediction.
+        rice: Boolean value, where true is rice (default), false is other species.
+        frame: the frame index, default value is 0, i.e., first frame.
+
+    Returns:
+        An array of points location in shape of (# points, 2).
+    """
+    if rice:
+        pts_lr = get_pts_pr_lr(h5, rice=True, frame=frame, primaryroot=False)
+        pts_all = pts_lr.reshape(-1, 2)
+    else:
+        pts_pr = get_pts_pr_lr(h5, rice=False, frame=frame, primaryroot=True)
+        pts_lr = get_pts_pr_lr(h5, rice=False, frame=frame, primaryroot=False)
+        primary_points = pts_pr.reshape(-1, 2)
+        lateral_points = pts_lr.reshape(-1, 2)
+        pts_all = np.concatenate((primary_points, lateral_points), axis=0)
+    return pts_all
```

Comment:
I don't like this organization -- it's going to result in loading the `Series` again for every frame.

If you need this type of logic, it should happen at the whole plant level (`get_traits_plant`). If it can't go in the loop in that function, maybe some of this should be in the `Series` class, like `Series.get_points(frame_idx)`.

I especially don't like hardcoding the names for the primary vs lateral suffixes (e.g., `"main_3do_6nodes"`) since these may change per dataset. We should let the `Series` class handle that logic.